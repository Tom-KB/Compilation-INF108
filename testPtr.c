void main(int arg) {
    int x;
    x = 5;
    int w;
    int* wp;
    w = 892;
    wp = &w;

    int* y;
    y = &x;
    print_int(*y);
    *y = 15;
    print_int(*y);
    print_int(x);
    x = 10;
    print_int(*y);
    print_int(x);

    int** z;
    z = &y;
    **z = 19;
    print_int(**z);
    print_int(*y);
    z = &wp;
    print_int(**z);
    print_int(x);
}