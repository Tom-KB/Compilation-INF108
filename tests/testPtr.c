int* y;

void lecteur(int j) {
    print_int(*(y+2)+j); // Lecture d'un entier dans une fonction à partir d'un pointeur
}

void main(int arg) {
    int x;
    x = 4;
    
    y = malloc(5*4);
    
    *(y+0) = x; // Assignation de la valeur d'une variable à un champ
    *(y+1) = 15;
    *(y+2) = 20;
    print_int(*(y+0));
    print_int(*(y+1));
    print_int(*(y+2));
    print_int(*(y+3)); // Non initialisée, vaut donc 0
    
    lecteur(2);
    
    int* m;
    m = malloc(5*4);

    int** g;
    g = malloc(4*1);

    int* k;

    g = &m;
    *(m + 0) = 5;
    *(m + 1) = 45;

    k = (m + 1);
    print_int(*k); 
    print_int(*(k-1)); // Equivaut à m

    print_int(k-m); // 1

    print_int(**(g + 0) + 2); 
    
    int i;
    i = 1;
    // Arithmétique avec variable
    print_int(*(m + i)); 
    print_int(*(k - i)); 
    
}