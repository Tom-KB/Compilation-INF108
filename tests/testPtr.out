4
15
20
0
22
45
5
1
7
45
5
